src/assert.rs

```diff
@@ -529,6 +647,52 @@ impl OutputAssertionBuilder {
     }
 }
 
+/// A type for writing to stdin during a test.
+pub trait InputPredicate {
```

Comment:
When we redo the API to use the predicates crate, this name will become confusing.  Maybe `StdinWriter`?

I'd say we could reuse the `Write` trait but that'd be annoying with providing trait implementations.  Maybe we don't make it so automatic but provide a helper `content` factory or something?